File: examples/border3/border3.rb
#!/usr/bin/ruby

## set user list
user_list = [
  { "name" => "sumire", "email" => "voilet@mail.com" },
  { "name" => "nana",   "email" => "seven@mail.com" },
  { "name" => "momoko", "email" => "peach@mail.com" },
  { "name" => "kasumi", "email" => "mist@mail.com" },
]

## display view
require 'erb'
filename = 'border3.view'
str = File.open(filename) { |f| f.read }  ## or File.read(filename)
trim_mode = 1
erb = ERB.new(str, $SAFE, trim_mode)
print erb.result(binding())

File: examples/border3/Border3.java
/*
 * @(#) Border3.java
 * @id  $Id$
 */
import java.util.*;
import java.io.*;
import kwartz.Kwartz;
import kwartz.Context;
import kwartz.Template;


public class Border3 {

    private static Kwartz kwartz = new Kwartz();
    private String pdataFilename;
    private String plogicFilename;

    public Border3() {
        this("border3.html", "border3.plogic");
    }

    public Border3(String pdataFilename, String plogicFilename) {
        this.pdataFilename  = pdataFilename;
        this.plogicFilename = plogicFilename;
    }

    public void execute(Writer writer) throws Exception {
        // set user list
        String[][] list = {
            new String[] { "sumire", "violet@mail.com", },
            new String[] { "nana",   "seven@mail.com", },
            new String[] { "momoko", "peach@mail.com", },
            new String[] { "kasumi", "mist@mail.com", },
        };
        List user_list = new ArrayList();
        for (int i = 0; i < list.length; i++) {
            String[] tuple = list[i];
            Map user = new HashMap();
            user.put("name",  tuple[0]);
            user.put("email", tuple[1]);
            user_list.add(user);
        }

        // compile template
        String charset = System.getProperty("file.encoding");
        String cacheKey = "border3";
        Template template = kwartz.getTemplate(cacheKey, pdataFilename, plogicFilename, null, charset);

        // execute
        Context context = new Context();
        context.put("user_list", user_list);
        template.execute(context, writer);
    }

    public static void main(String[] args) {
        try {
            Writer writer = new OutputStreamWriter(System.out);
            Border3 border3 = new Border3();
            border3.execute(writer);
            writer.flush();
        } catch (Exception ex) {
            ex.printStackTrace();
        }
    }

} // class end
